wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/event/MachineStateListener.java

```diff
@@ -62,6 +70,12 @@ public void onEvent(MachineStatusEvent event) {
             } catch (NotFoundException | ServerException | ConflictException exception) {
                 LOG.error(exception.getLocalizedMessage(), exception);
             }
+        } else if (CREATING.equals(event.getEventType()) && !event.isDev()) {
+            try {
+                workspaceManager.getWorkspace(workspaceId).getRuntime().addMachine(machineManager.getMachine(event.getMachineId()));
```

Comment:
It will not affect the runtime instance, see [javadoc](https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java#L95-L100)